test_pet.py

```diff
@@ -19,28 +21,51 @@ def test_pet_schema():
     # Validate the response schema against the defined schema in schemas.py
     validate(instance=response.json(), schema=schemas.pet)
 
+
 '''
 TODO: Finish this test by...
 1) Extending the parameterization to include all available statuses
 2) Validate the appropriate response code
 3) Validate the 'status' property in the response is equal to the expected status
 4) Validate the schema for each object in the response
 '''
-@pytest.mark.parametrize("status", [("available")])
+
+'''
+QA_NOTE: Empty records returned for "sold" status but it is returning 200. 
+Not sure may be the expectation is to validate what status available for users
+'''
+
+
+@pytest.mark.parametrize("status", ["available", "pending", "sold"])
 def test_find_by_status_200(status):
     test_endpoint = "/pets/findByStatus"
     params = {
         "status": status
     }
 
     response = api_helpers.get_api_data(test_endpoint, params)
-    # TODO...
+    assert response.status_code == 200
+    for res in response.json():
+        validate(instance=res, schema=schemas.pet)
+        assert res["status"] == status
+
 
 '''
 TODO: Finish this test by...
 1) Testing and validating the appropriate 404 response for /pets/{pet_id}
 2) Parameterizing the test for any edge cases
 '''
-def test_get_by_id_404():
-    # TODO...
-    pass
\ No newline at end of file
+
+
+'''
+QA OBSERVATION: The test is passing for "a_str" as path. 
```

Comment:
good callout, i would consider this a bug, a different response is appropriate here